nbgrader/api.py

```diff
@@ -7,11 +7,10 @@
 
 from sqlalchemy import (create_engine, ForeignKey, Column, String, Text,
                         DateTime, Interval, Float, Enum, UniqueConstraint,
-                        Boolean, inspect)
+                        Boolean, inspect, text)
 from sqlalchemy.orm import (sessionmaker, scoped_session, relationship,
-                            column_property)
+                            column_property, declarative_base)
```

Comment:
The preferred method for 2.0 is to use the `DeclarativeBase` class. This still works, but switching to the class makes a lot of sense in this PR.
see: https://docs.sqlalchemy.org/en/20/orm/mapping_api.html#sqlalchemy.orm.declarative_base

Reply:
Thank you for letting me know. I don't recall seeing a mention of this in the migration document (maybe I missed it), but I agree that it makes sense to switch. Will fix accordingly.

Reply:
@shreve @brichet `DeclarativeBase` is apparently a sqlalchemy 2.x addition (without being present in 1.4), so I should have bumped up the minimum version. 

The previous `declarative_base` imported from `sqlalchemy.orm` should work with both 1.4 and 2.x.

The difference between them [is](https://docs.sqlalchemy.org/en/20/orm/mapping_api.html#sqlalchemy.orm.DeclarativeBase) for type-checkers, which at least right now is not a big deal for `nbgrader`. So I think it is worth going back to `declarative_base` over `DeclarativeBase`. What do you think?